hipe4ml/tree_handler.py

```diff
@@ -0,0 +1,278 @@
+        self._projection_binning = None
+        self._sliced_df_list = None
+
+    def get_data_frame(self):
+        """
+        Get the pandas DataFrame stored in the TreeHandler
+
+        Returns
+        ------------------------------------------------
+        out: pandas.DataFrame
+            DataFrame stored in the TreeHandler
+        """
+        return self._full_data_frame
+
+    def get_preselections(self):
+        """
+        Get the preselections applied to the stored DataFrame
+
+        Returns
+        ------------------------------------------------
+        out: str
+            String containing the cuts applied to the stored DataFrame
+
+        """
+        return self._preselections
+
+    def get_projection_variable(self):
+        """
+        Get the name of the sliced variable
+
+        Returns
+        ------------------------------------------------
+        out: str
+            Sliced variable
+
+        """
+        return self._projection_variable
+
+    def get_projection_binning(self):
+        """
+        Get the bins used for slicing the DataFrame
+
+        Returns
+        ------------------------------------------------
+        out: list
+            Each element of the list is a list containing the
+            bin edges
+
+        """
+        return self._projection_binning
+
+    def get_slice(self, n_bin):
+        """
+        Get the n-th slice of the original DataFrame
+
+        Parameters
+        ------------------------------------------------
+        n_bin: int
+            n-th element of _projection_binning list.
+
+        Returns
+        ------------------------------------------------
+        out: pandas.DataFrame
+            N-th Slice of the original DataFrame
+        """
+        return self._sliced_df_list[n_bin]
+
+    def get_sliced_df_list(self):
+        """
+        Get the list containing the slices of the orginal
+        DataFrame
+
+        Returns
+        ------------------------------------------------
+        out: list
+            List containing the slices of the orginal
+            DataFrame
+
+        """
+        return self._sliced_df_list
+
+    def apply_preselections(self, preselections, inplace=True):
+        """
+        Apply preselections to data
+
+        Parameters
+        ------------------------------------------------
+        preselection: str
+            String containing the cuts to be applied as preselection on the data contained in the original
+            tree. The string syntax is the one required in the pandas.DataFrame.query() method.
+            You can refer to variables in the environment by prefixing them with an ‘@’ character like @a + b.
+        inplace: bool
+            If True, the preselected dataframe replaces the initial dataframe. Otherwise return a copy of the
+            preselected df
+
+        Returns
+        ------------------------------------------------
+        out: pandas.DataFrame or None
```

Comment:
I think it is better to keep the names used in the code, this is what scikit-learn does for example.